frontend/src/component/ui/ChangeLogModal.tsx

```diff
@@ -25,6 +31,23 @@ export class ChangeLogModal extends Component<Props, State> {
     shouldShowFullChangeLog: false
   };
 
+  private modalRef = React.createRef<Modal>();
+
+  componentDidMount() {
+    if (this.props.shouldShowModal) {
+      this.open();
+    }
+  }
+
+  componentDidUpdate(prevProps: Props) {
+    if (
+      this.props.shouldShowModal !== prevProps.shouldShowModal &&
+      this.props.shouldShowModal
+    ) {
+      this.open();
+    }
```

Comment:
```suggestion
    if (this.props.shouldShowModal === this.props.shouldShowModal) {
        return;
    }
    if (!this.props.shouldShowModal) {
        return;
    }
    this.open();
```
I think we should let the owner of the modal to decide whether or when to open it. We leave the automatic opening logic inside this modal.

Reply:
Hi @kapillamba4, I mis typed the comment. I am so sorry.